pkg/controller/common/helpers.go

```diff
@@ -125,6 +125,13 @@ func MergeMachineConfigs(configs []*mcfgv1.MachineConfig, osImageURL string) (*m
 		}
 	}
 
+	// For layering, we want to let the user override OSImageURL again
+	for _, cfg := range configs {
+		if cfg.Spec.OSImageURL != "" {
+			osImageURL = cfg.Spec.OSImageURL
```

Comment:
Just curious, is our current plan to override the base OSImageURL as opposed to a new field?

minor nit: there is a comment at the start of the function 
`// It uses only the OSImageURL provided by the CVO and ignores any MC provided OSImageURL.`
we should probably remove

Reply:
Maybe I let Colin twist my arm too hard, but yes, the current plan was to let the user "take the wheel" and override OSImageURL. :smile:

I updated the comment to match this PR's behavior.